### PR TITLE
Handle the error for private helm repo

### PR DESCRIPTION
**Pull request by StrongMonkey**
https://github.com/rancher/fleet/issues/340
https://github.com/rancher/fleet/issues/341

### CODE REVIEW

File: pkg/bundle/resources.go [hunk 1]
```diff
@@ -135,6 +135,10 @@ func chartURL(location *fleet.HelmOptions, auth Auth) (string, error) {
 		return "", err
 	}
 
+	if resp.StatusCode != 200 {
+		return "", fmt.Errorf("failed to read helm repo from %s, error code: %v, response body: %s", location.Repo+"index.yaml", resp.StatusCode, bytes)
```
Comment (nickgerace) [inline pkg/bundle/resources.go:139]: Are bytes always type `string`? Or can it be a slice of bytes?

File: pkg/controllers/controllers.go [hunk 2]
```diff
@@ -188,7 +188,8 @@ func Register(ctx context.Context, systemNamespace string, cfg clientcmd.ClientC
 		appCtx.BundleDeployment(),
 		appCtx.GitRepoRestriction().Cache(),
 		appCtx.Bundle(),
-		appCtx.GitRepo())
+		appCtx.GitRepo(),
+		appCtx.Core.Secret().Cache())
```
Comment (nickgerace) [inline pkg/controllers/controllers.go:192]: Why cache the secret?

Comment (StrongMonkey) [reply pkg/controllers/controllers.go:192]: This avoids api call which cause latency

File: pkg/bundle/resources.go (hunk 1 above)
Comment (StrongMonkey) [reply pkg/bundle/resources.go:139]: yeah, doing `%s` with bytes will also work

Comment (StrongMonkey) [reply pkg/bundle/resources.go:139]: https://golang.org/pkg/fmt/ good catch :+1: I checked the reference and it works with string/slices of bytes

**Review by nickgerace — APPROVED** [review]

**Pull request merged**